### ISSUE TITLE
Confirmation before reset data in content manager

### ISSUE THREAD

**Issue by oeddyo** [strapi/strapi#1039]
Hi Strapi,

We are starting to use strapi (like seriously use it) in my company. One complaint from our ops people is (since they are using it as CMS in admin dashboard) there's no confirmation for "Reset" button and because of mis-conduction they lost a couple of hours worth of work : | 

Wondering if it's reasonable to add a confirmation before dangerous ops like this 

Thanks!

![image](https://user-images.githubusercontent.com/1943146/39289715-d7581c30-495f-11e8-81f8-a0ac456119d9.png)

---

- [Y] I'm sure that this issue hasn't already been referenced

**Comment by lauriejim** [comment]
Hello @oeddyo I think it's an easy thing to add on Strapi. 
You can make this update by yourself and submit a PR.

@soupette Can help you to make it.

**Comment by soupette** [comment]
Hi @oeddyo 

I'm not sure we're going to develop such feature however here's how you can do it.

First create a project in dev mode ()[plugin development](https://strapi.io/documentation/plugin-development/quick-start.html)).

In the `content-manager/admin/src/containers/index.js` you can make the following modifications:

```
import PopUpWarning from 'components/PopUpWarning';

export class EditPage extends React.Component {
  state = { showWarning: false };

  // ...

  toggleModalWarning = () => this.setState(prevState => ({ showWarning: !prevState.showWarning }));

  pluginHeaderActions =  () => (
    [
      {
        label: 'content-manager.containers.Edit.reset',
        kind: 'secondary',
        onClick: () => this.setState({ showWarning: true }),
        type: 'button',
      },
      {
        kind: 'primary',
        label: 'content-manager.containers.Edit.submit',
        onClick: this.handleSubmit,
        type: 'submit',
        loader: this.props.editPage.showLoader,
        style: this.props.editPage.showLoader ? { marginRight: '18px' } : {},
      },
    ]
  );

  render() {
    return (
      <div>
        <PopUpWarning
          isOpen={this.state.showWarning}
          toggleModal={this.toggleModalWarning}
          content={{ message: 'yourTranslation' }}
          popUpWarningType={'danger'}
          onConfirm={this.props.onCancel}
        />
      </div>
    );
  }
}
```

You'll also have to create the associated translations in the plugin's translations folder.

Feel free to submit a PR if you get it working.

**Comment by lauriejim** [comment]
@oeddyo feel free to follow @soupette instruction to submit a PR. Will appreciate your contribution.

Check out the contributing guide to get started: https://github.com/strapi/strapi/blob/master/.github/CONTRIBUTING.md

**Comment by derrickmehaffy** [comment]
@oeddyo any update on this? Just checking to see if we can close out old issues. I'm sure the team would really love a PR (they are busy guys)

**Comment by lauriejim** [comment]
Can you please give us an update about this issue? Did you work on it @oeddyo following @soupette code ?

**Comment by soupette** [comment]
@lauriejim https://github.com/strapi/strapi/commit/0b1a280c7aa96742281c9b0ba6bd7c0739f98ee3#diff-a1018c08b7910d0259b23d23ec7a9fb9

**Comment by lauriejim** [comment]
Thank you @soupette

**Comment by soupette** [comment]
I'm closing this issue it has been implemented in #1529
